src/box/box.cc

```diff
@@ -4346,6 +4357,8 @@ box_cfg_xc(void)
 
 	rmean_cleanup(rmean_box);
 
+	box_run_on_recovery_state(RECOVERY_STATE_WAL_RECOVERED);
```

Comment:
Let's do this in `boostrap_master`, `bootstrap_from_master`, and `local_recovery` - the code will be easier to follow if "wal_recovered" is located after "snapshot_recovered" in the same function.

Reply:
Done. I've also simplifed the bootstrap sequence a bit.
<details><summary>diff</summary>

```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index 4eafe0d3a..363452f30 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -3779,6 +3779,11 @@ bootstrap_master(const struct tt_uuid *replicaset_uuid)
                panic("failed to create a checkpoint");
 
        box_run_on_recovery_state(RECOVERY_STATE_SNAPSHOT_RECOVERED);
+       box_run_on_recovery_state(RECOVERY_STATE_WAL_RECOVERED);
+       box_run_on_recovery_state(RECOVERY_STATE_INDEXES_BUILT);
+       assert(!recovery_state_synced_is_reached);
+       box_run_on_recovery_state(RECOVERY_STATE_SYNCED);
+       recovery_state_synced_is_reached = true;
 }
 
 /**
@@ -3864,6 +3869,8 @@ bootstrap_from_master(struct replica *master)
        if (gc_checkpoint() != 0)
                panic("failed to create a checkpoint");
 
+       box_run_on_recovery_state(RECOVERY_STATE_WAL_RECOVERED);
+
        return true;
 }
 
@@ -4164,6 +4171,8 @@ local_recovery(const struct tt_uuid *instance_uuid,
                          tt_uuid_str(replicaset_uuid),
                          tt_uuid_str(&REPLICASET_UUID));
        }
+
+       box_run_on_recovery_state(RECOVERY_STATE_WAL_RECOVERED);
 }
 
 static void
@@ -4362,8 +4371,6 @@ box_cfg_xc(void)
 
        rmean_cleanup(rmean_box);
 
-       box_run_on_recovery_state(RECOVERY_STATE_WAL_RECOVERED);
-
        /* Follow replica */
        replicaset_follow();
 
@@ -4401,12 +4408,6 @@ box_cfg_xc(void)
        title("running");
        say_info("ready to accept requests");
 
-       if (is_bootstrap_leader) {
-               box_run_on_recovery_state(RECOVERY_STATE_INDEXES_BUILT);
-               assert(!recovery_state_synced_is_reached);
-               box_run_on_recovery_state(RECOVERY_STATE_SYNCED);
-               recovery_state_synced_is_reached = true;
-       }
        if (!is_bootstrap_leader) {
                replicaset_sync();
        } else if (box_election_mode == ELECTION_MODE_CANDIDATE ||
```

</details>